### ISSUE TITLE
[object_detection] Using Adam Optimizer 

### ISSUE THREAD

**Issue by gustavz** [tensorflow/models#4418]
### System information
- **What is the top-level directory of the model you are using**: object_detection
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes and No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:  Linux Ubuntu 16.04)
- **TensorFlow installed from (source or binary)**: Source 
- **TensorFlow version (use command below)**: 1.7
- **Python version**:  2.7
- **Bazel version (if compiling from source)**: 0.13.0
- **GCC/Compiler version (if compiling from source)**: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.9) 5.4.0 20160609
- **CUDA/cuDNN version**: 9 / 7
- **GPU model and memory**: 4GB GTX 1050
- **Exact command to reproduce**:

in the object_detection/protos only 3 optimizers are supported (Adam,Momentum,RMS-Prop)
As for my understanding Adam is the most advanced. If this is true why is it not used in any model of the model_zoo? All Faster RCNN and Mask RCNN Models use Momentum (which is, what i thought, outdated).

- So why not use Adam? Is there any disadvantage in using adam with those bigger models?
- Did anybody ever use Adam instead and check the results?
- If i want to use Adam to train a Mask RCNN Model, how should i set the LEarningRate? Constant? Scheduled? ExponentiallyDecayed? Because  as i understood Adam it adjusts LR by itself, but this implementation does not seem to do it.
- Why is it only possible to pass LearninRate (as defined in the protos)? Why not also the other Adam specific parameters?

**Comment by bitfort** [comment]
Adding someone who can help.

**Comment by giacomobartoli** [comment]
I'm interested in the same discussion. I would like to have a .config example using Adam as optimizer.
Thanks @GustavZ for point it out.

**Comment by giacomobartoli** [comment]
@GustavZ I asked the same question on SOF. I think the answer they gave me could help you:
https://stackoverflow.com/questions/51915803/tensorflow-object-detection-use-adam-instead-of-rmsprop/51920195#51920195

**Comment by MichaelW17** [comment]
I don't think these questions are well answered by now in this post.

**Comment by sohinimallick** [comment]
Are there any developments in this?

**Comment by vighneshbirodkar** [comment]
The CenterNet models use the Adam optimizer
https://github.com/tensorflow/models/blob/master/research/object_detection/configs/tf2/centernet_hourglass104_1024x1024_coco17_tpu-32.config

1e-3 is usually a good starting point for Adam, although in literature people have used other values. The optimizer each model uses really depends on the literature. In most cases we have stuck to something similar to what the original paper has used.

I am closing this for now since this isn't really an issue and with CenterNet, we have examples of using Adam.
